.github/workflows/publish-release.yml

```
@@ -0,0 +1,56 @@
+name: Publish final artifacts to Maven Central repository
+
+on:
+  push:
+    tags:
+      - armeria-*
+
+jobs:
+  publish:
+    name: Publish final artifacts
+    if: github.repository == 'line/armeria'
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v2
+
+      - id: setup-jdk-16
+        name: Set up JDK 16
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '16'
+
+      - name: Restore the cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/wrapper
+            ~/.gradle/caches
+          key: build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('gradle.properties', 'gradle/wrapper/gradle-wrapper.properties', '**/build.gradle', 'dependencies.yml', '*/package-lock.json') }}
+          restore-keys: |
+            build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-
+            build-${{ matrix.java }}-${{ runner.os }}-
+
+      - name: Build with Gradle
+        run: |
+          ./gradlew --no-daemon --stacktrace --max-workers=8 --parallel -PflakyTests=false build
+        shell: bash
+
+      - name: Publish and close repository
+        run: |
+          ./gradlew --no-daemon --stacktrace --max-workers=1 clean publish closeAndReleaseStagingRepository
```

```suggestion
          ./gradlew --no-daemon --stacktrace --max-workers=1 publish closeAndReleaseStagingRepository
```

Don't think it's common to clean+rebuild the tested artifacts before publishing them